### PR TITLE
[IMP] point_of_sale: hide sync notification icon when operational

### PR DESCRIPTION
**Pull request by AH-Yussef**
No longer show sync notification icon when operational/green (wifi icon)

### before
Screenshot 2024-07-16 at 6 11 09 PM

### after
Screenshot 2024-07-16 at 6 07 34 PM

task-4045935

### CODE REVIEW

File: addons/point_of_sale/static/tests/tours/utils/product_screen_util.js [hunk 1]
```diff
@@ -539,10 +539,6 @@ export function finishOrder() {
             trigger: ".payment-screen .btn-switchpane:contains('Validate')",
             run: "click",
         },
-        {
-            content: "verify that the order has been successfully sent to the backend",
-            trigger: ".js_connected:visible",
-        },
```
Comment (vlst-odoo) [inline addons/point_of_sale/static/tests/tours/utils/product_screen_util.js:545]: here you probably want to make sure that the icon is not shown

File: addons/point_of_sale/static/src/app/navbar/sync_notification/sync_notification.xml [hunk 2]
```diff
@@ -15,10 +15,7 @@
                 <i class="fa fa-fw fa-spin fa-circle-o-notch" role="img"
                    aria-label="Synchronisation Connecting" title="Synchronisation Connecting"></i>
             </div>
-            <div t-else="" class="js_connected oe_icon text-success">
-               <i class="fa fa-fw fa-wifi" role="img" aria-label="Synchronisation Connected"
-                   title="Synchronisation Connected"></i>
-            </div>
+            <div t-else="" class="js_connected oe_icon text-success" style="visibility: hidden;"/>
```
Comment (vlst-odoo) [inline addons/point_of_sale/static/src/app/navbar/sync_notification/sync_notification.xml:18]: can't you remove this line ?

File: addons/point_of_sale/static/src/app/navbar/sync_notification/sync_notification.xml [hunk 3]
```diff
@@ -2,7 +2,7 @@
 <templates id="template" xml:space="preserve">
 
     <t t-name="point_of_sale.SyncNotification">
-        <div class="oe_status btn btn-light rounded-0" t-on-click="onClick">
+        <div class="oe_status btn btn-light rounded-0" t-att-style="!(sync.offline || sync.loading) ? 'padding: unset;' : ''" t-on-click="onClick">
```
Comment (vlst-odoo) [inline addons/point_of_sale/static/src/app/navbar/sync_notification/sync_notification.xml:5]: can you find a way to not need the `t-att-style` ?

**Review by vlst-odoo — CHANGES_REQUESTED** [review]

File: addons/point_of_sale/static/src/app/navbar/sync_notification/sync_notification.xml (hunk 2 above)
Comment (AH-Yussef) [reply addons/point_of_sale/static/src/app/navbar/sync_notification/sync_notification.xml:18]: It's needed for this check https://github.com/odoo/odoo/blob/dee10df5c180ef376f4d334e18e4ee2de562ffe6/addons/pos_restaurant/static/tests/tours/pos_restaurant_tour.js#L16-L23

Comment (vlst-odoo) [reply addons/point_of_sale/static/src/app/navbar/sync_notification/sync_notification.xml:18]: before, the way to see that you are connected was to look for the green icon; now, as long as you don't see the "disconnected" icon, it means that you are connected, so you can change the tours to check that you don't see the "disconnected" icon instead

Comment (vlst-odoo) [reply addons/point_of_sale/static/src/app/navbar/sync_notification/sync_notification.xml:18]: before, the way to see that you are connected was to look for the green icon; now, as long as you don't see the "disconnected" icon, it means that you are connected, so you can change the tours to check that you don't see the "disconnected" icon instead

File: addons/point_of_sale/static/tests/tours/utils/common.js [hunk 4]
```diff
@@ -57,3 +57,9 @@ export function selectButton(name) {
         run: "click",
     };
 }
+
+export function isSyncStatusConnected() {
+    return {
+        trigger: `.pos-topheader .pos-rightheader ${negate(".oe_status", ".status-buttons")}`,
```
Comment (vlst-odoo) [inline addons/point_of_sale/static/tests/tours/utils/common.js:63]:
```suggestion
        trigger: negate(".oe_status", ".pos-rightheader.status-buttons"),
```

File: addons/point_of_sale/static/src/app/navbar/navbar.xml [hunk 5]
```diff
@@ -24,7 +24,21 @@
                     <CashierName />
                     <ProxyStatus t-if="pos.config.use_proxy" />
                     <button t-if="customerFacingDisplayButtonIsShown" t-on-click="openCustomerDisplay" class="btn btn-light fa fa-fw fa-desktop text-success pe-4" title="Customer Display" />
-                    <SyncNotification />
+                    
+                    <div t-if="1" class="oe_status btn btn-light rounded-0 d-flex gap-1" t-on-click="onSyncNotificationClick">
```
Comment (vlst-odoo) [inline addons/point_of_sale/static/src/app/navbar/navbar.xml]: there is no need of the `t-if` here

File: addons/point_of_sale/static/tests/tours/utils/common.js [hunk 6]
```diff
@@ -57,3 +57,9 @@ export function selectButton(name) {
         run: "click",
     };
 }
+
+export function isSyncStatusConnected() {
+    return {
+        trigger: negate(".oe_status", ".pos-rightheader .status-buttons"),
+    };
+}
```
Comment (vlst-odoo) [inline addons/point_of_sale/static/tests/tours/utils/common.js:65]: i think that it makes more sense to  have this in `chrome_util.js`

File: addons/point_of_sale/static/src/app/navbar/navbar.xml [hunk 7]
```diff
@@ -24,7 +24,17 @@
                     <CashierName />
                     <ProxyStatus t-if="pos.config.use_proxy" />
                     <button t-if="customerFacingDisplayButtonIsShown" t-on-click="openCustomerDisplay" class="btn btn-light fa fa-fw fa-desktop text-success pe-4" title="Customer Display" />
-                    <SyncNotification />
+                    
+                    <div t-if="this.pos.data.network.offline" class="oe_status btn btn-light rounded-0 d-flex gap-1" t-on-click="onSyncNotificationClick">
+                        <t t-set="pending" t-value="this.pos.data.network.unsyncData.length" />
```
Comment (vlst-odoo) [inline addons/point_of_sale/static/src/app/navbar/navbar.xml:29]: you can put this outside of the div so that you don't have to duplicate it

**Review by vlst-odoo — CHANGES_REQUESTED** [review]

File: addons/point_of_sale/static/src/app/navbar/navbar.xml [hunk 8]
```diff
@@ -24,7 +24,16 @@
...
+                    
+                    <t t-set="pending" t-value="this.pos.data.network.unsyncData.length" />
+                    <div t-if="this.pos.data.network.offline" class="oe_status btn btn-light rounded-0 d-flex gap-1" t-on-click="onSyncNotificationClick">
+                        <span t-if="pending > 0" t-esc="pending" /> 
+                        <div class="fa fa-fw fa-chain-broken text-danger"/>
+                    </div>
+                    <div t-elif="this.pos.data.network.loading" class="oe_status btn btn-light rounded-0 d-flex gap-1" t-on-click="onSyncNotificationClick">
+                        <span t-if="pending > 0" t-esc="pending" />                        
+                        <div class="fa fa-fw fa-spin fa-circle-o-notch text-warning"/>
+                    </div>
```
Comment (vlst-odoo) [inline addons/point_of_sale/static/src/app/navbar/navbar.xml:36]:
```suggestion
                    <div t-if="pos.data.network.offline or pos.data.network.loading" class="oe_status btn btn-light rounded-0 d-flex gap-1" t-on-click="onSyncNotificationClick">
                        <span t-if="pos.data.network.unsyncData.length > 0" t-esc="pos.data.network.unsyncData.length" /> 
                        <div t-if="pos.data.network.offline" class="fa fa-fw fa-chain-broken text-danger"/>
                        <div t-else="" class="fa fa-fw fa-spin fa-circle-o-notch text-warning"/>
                    </div>
```

**Review by vlst-odoo — APPROVED** [review]
robodoo r+

**Pull request closed without merging**